app/controllers/account/companies_controller.rb

```diff
@@ -16,6 +16,7 @@ def new
   def create
     @company = current_user.companies.build(company_params)
     if @company.save
+      BingApiV7.new.bing_pages_to_model(@company)
```

Comment:
you need to pass @company in initialize, and just call "process" method, or something like this